### PR TITLE
EZP-30427: Changed drafts pagination to use limit and offset

### PR DESCRIPTION
**Pull request by mikadamczyk**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-30427
| Bug fix?      | yes
| New feature?  |no
| BC breaks?    | no
| Tests pass?   |no
| Doc needed?   |no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

This PR changed thy way which pagination for content drafts list load objects. The current implementation uses offset and limit to improve performance when many drafts are fetched (more than 5000). That change affects Dashboard and  list of user's drafts (`admin/contentdraft/list`)

If a user has no access to `content/versionread` then he will see information like `User does not have access to 'versionread' 'content' with content ID: 5178`

Related PR: https://github.com/ezsystems/ezpublish-kernel/pull/2753

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/lib/Pagination/Pagerfanta/ContentDraftAdapter.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Pagination\Pagerfanta;
+
+use eZ\Publish\API\Repository\BookmarkService;
+use eZ\Publish\API\Repository\ContentService;
+use EzSystems\EzPlatformAdminUi\UI\Dataset\DatasetFactory;
+use Pagerfanta\Adapter\AdapterInterface;
+
+class ContentDraftAdapter implements AdapterInterface
```
Comment (adamwojs) [inline src/lib/Pagination/Pagerfanta/ContentDraftAdapter.php]:
```suggestion
final class ContentDraftAdapter implements AdapterInterface
```

**Comment by mikadamczyk** [comment]
Travis: https://travis-ci.org/ezsystems/ezplatform-admin-ui/builds/583165957

File: src/bundle/Resources/views/admin/content_draft/draft_list.html.twig [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                                    <th class="ez-table__header-cell ez-table__header-cell--after-icon">{{ 'drafts.list.name'|trans|desc('Name') }}</th>
+                                    <th class="ez-table__header-cell">{{ 'drafts.list.content_type'|trans|desc('Content Type') }}</th>
+                                    <th class="ez-table__header-cell">{{ 'drafts.list.modified_language'|trans|desc('Modified Language') }}</th>
+                                    <th class="ez-table__header-cell">{{ 'drafts.list.version'|trans|desc('Version') }}</th>
+                                    <th class="ez-table__header-cell">{{ 'drafts.list.last_saved'|trans|desc('Last Saved') }}</th>
+                                    <th class="ez-table__header-cell"></th>
+                                </tr>
+                            </thead>
+                            <tbody>
+                            {% if pager.count is same as(0) %}
```
Comment (alongosz) [inline src/bundle/Resources/views/admin/content_draft/draft_list.html.twig]: This construction looks quite weird to me. Could you use `for-else` instead? Ref.: https://twig.symfony.com/doc/2.x/tags/for.html#the-else-clause

File: src/lib/Pagination/Pagerfanta/ContentDraftAdapter.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    /**
+     * Returns the number of results.
+     *
+     * @return int the number of results
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    public function getNbResults()
```
Comment (alongosz) [inline src/lib/Pagination/Pagerfanta/ContentDraftAdapter.php]:
```suggestion
    public function getNbResults(): int
```

File: src/lib/Pagination/Pagerfanta/ContentDraftAdapter.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+     * Returns an slice of the results.
+     *
+     * @param int $offset the offset
+     * @param int $length the length
+     *
+     * @return array|\Traversable the slice
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    public function getSlice($offset, $length)
```
Comment (alongosz) [inline src/lib/Pagination/Pagerfanta/ContentDraftAdapter.php]:
```suggestion
    public function getSlice($offset, $length): \Traversable
```
or
```suggestion
    public function getSlice($offset, $length): array
```
depending on the actual context & invoker usage

File: src/lib/UI/Value/Content/ContentDraft.php [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\UI\Value\Content;
+
+use eZ\Publish\API\Repository\Values\Content\VersionInfo;
+use eZ\Publish\API\Repository\Values\ContentType\ContentType;
+
+class ContentDraft implements ContentDraftInterface
```
Comment (alongosz) [inline src/lib/UI/Value/Content/ContentDraft.php]: IMHO it's an overkill to create wrappers for VOs from Public API. Is it a pattern in AdminUI? Not requesting changes, just asking.

File: src/lib/UI/Dataset/ContentDraftListDataset.php [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+                return $this->valueFactory->createUnauthorizedContentDraft($contentDraftListItem);
+            },
+            $contentDraftListItems
+        );
+
+        return $this;
+    }
+
+    /**
+     * @return array
```
Comment (adamwojs) [inline src/lib/UI/Dataset/ContentDraftListDataset.php]:
```suggestion
     * @return \EzSystems\EzPlatformAdminUi\UI\Value\Content\ContentDraftInterface[]
```

File: src/lib/UI/Dataset/ContentDraftListDataset.php [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    /** @var \eZ\Publish\API\Repository\ContentService */
+    private $contentService;
+
+    /** @var \eZ\Publish\API\Repository\ContentTypeService */
+    private $contentTypeService;
+
+    /** @var \EzSystems\EzPlatformAdminUi\UI\Value\ValueFactory */
+    private $valueFactory;
+
+    /** @var array */
```
Comment (adamwojs) [inline src/lib/UI/Dataset/ContentDraftListDataset.php]:
```suggestion
    /** @var \EzSystems\EzPlatformAdminUi\UI\Value\Content\ContentDraftInterface[] */
```

**Comment by mikadamczyk** [comment]
ping @alongosz

**Pull request merged**